crypto/ec/ec_lib.c

```diff
@@ -487,6 +509,27 @@ int EC_GROUP_check_discriminant(const EC_GROUP *group, BN_CTX *ctx)
     return group->meth->group_check_discriminant(group, ctx);
 }
 
+static ossl_inline int pnt_is_compat(const EC_POINT *point,
+                                     const EC_GROUP *group)
+{
+    if ((group->meth != point->meth
+            && group->meth->point_cmp != point->meth->point_cmp)
+            || (group->curve_name != 0
+            && point->curve_name != 0
+            && group->curve_name != point->curve_name))
+        return 0;
+
+    return 1;
+}
+
+static int point_cmp(const EC_GROUP *group, const EC_POINT *a, const EC_POINT *b,
```

Comment:
This is completely superseding `EC_POINT_cmp()`. Why? Should we instead improve `EC_POINT_cmp()` to make sure the additional checks in `pnt_is_compat()` are always executed?

Reply:
I wasnt confident that applying that change everywhere would be good, it should be ok for the group_cmp. If the original writer of this code has an opinion on this, it would be good to know.